### ISSUE TITLE
Segmentation fault (core dumped) on font save

### ISSUE THREAD

**Issue by xb0ta** [johanmattssonm/birdfont#123]
Greetings, I really like Birdfont compared to other linux programs for developing fonts because the interface is much simpler. Unfortunately, after delving it for some time, I did not notice that the files I have been saving are incomplete (missing font info). I have tried invoking it in cli and the errors are as follows:

** (birdfont:10747): WARNING **: 22:16:35.228: TrackTool.vala:542: No path.
Can not save settings. (Error removing file /home/<user>/.config/birdfont: Directory not empty)Can not save settings. (Error removing file /home/<user>/.config/birdfont: Directory not empty)
** (birdfont:10747): WARNING **: 22:16:39.897: TrackTool.vala:542: No path.
Can not save settings. (Error removing file /home/<user>/.config/birdfont: Directory not empty)Can not save settings. (Error removing file /home/<user>/.config/birdfont: Directory not empty)
** (birdfont:10747): WARNING **: 22:16:43.186: TrackTool.vala:542: No path.

** (birdfont:10747): WARNING **: 22:16:43.186: TrackTool.vala:542: No path.

** (birdfont:10747): WARNING **: 22:16:43.992: EditPoint.vala:388: EditPoint.next is null

** (birdfont:10747): CRITICAL **: 22:16:43.992: __lambda20_: assertion 'b != NULL' failed
Segmentation fault (core dumped)

Output file generated is attached below.
[font.txt](https://github.com/johanmattssonm/birdfont/files/7109292/ads.txt)

OS: Linux 
Distribution: Manjaro
Arch: 64bit

**Comment by johanmattssonm** [comment]
Thank you for the report. Do you have a .bf file that I can reproduce his or do you happen to know what to click on in order for this to happen?

**Comment by xb0ta** [comment]
Thank you for the prompt response, I have attached the birdfont file in my previous comment as font.txt (that is the generated output on save). I do not think there is an order for this to happen since the first time I initialised Birdfont the error started after I saved my first font. I do believe this is either a build malfunction ( from the maintainer's version in the arch repo) or a new bug in this version of Birdfont.

Reattached file saved as archived:
[output.zip](https://github.com/johanmattssonm/birdfont/files/7110625/ads.birdfont.zip)

**Comment by johanmattssonm** [comment]
Thank you. I have spent some time trying to reproduce this but I could not get the save function to crash. Please let me know if you find a way to make this happen.

**Comment by xb0ta** [comment]
After spending hours of debugging , I have concluded that it was actually the maintainer's build from the arch repository  that caused the segmentation fault. From the official birdfont.org website I downloaded the source and built it from scratch using flatpak and voila it worked. The program now works spendidly. Sorry for the bother, should have tried building from source sooner. Everything is doing great, thumbs up.

Solution for Manjaro/Arch users:

Build the program from source via the official birdfont website.
